src/public-api.rs

```diff
@@ -0,0 +1,195 @@
+//! Public API for the svc-cargo server
+//! REST Endpoints for client applications
+
+// use std::{
+//     sync::Arc,
+//     // collections::HashMap
+// };
+
+use chrono::naive::NaiveDateTime;
+
+use axum::{extract::Query, response::IntoResponse, Json};
+use hyper::{HeaderMap, StatusCode};
+use serde::{Deserialize, Serialize};
+use utoipa::{IntoParams, ToSchema};
+
+/// Flight Query
+#[allow(dead_code)]
+#[derive(Debug, Deserialize, IntoParams, Clone)]
+pub struct FlightQuery {
+    vport_depart_id: String,
+    vport_arrive_id: String,
+    timestamp: NaiveDateTime,
+    weight_kg: f32,
+}
+
+impl FlightQuery {
+    /// Creates a new flight query with required fields
+    /// # Arguments
+    /// vport_depart_id: The String ID of the vertiport to leave from
+    /// vport_arrive_id: The String ID of the destination vertiport
+    /// timestamp: The delivery time
+    /// weight_kg: The approximate weight of the cargo
+    pub fn new(
+        vport_depart_id: String,
+        vport_arrive_id: String,
+        timestamp: NaiveDateTime,
+        weight_kg: f32,
+    ) -> Self {
+        FlightQuery {
+            vport_depart_id,
+            vport_arrive_id,
+            timestamp,
+            weight_kg,
+        }
+    }
+}
+
+/// Region Query
+#[derive(Debug, Deserialize, IntoParams, Copy, Clone)]
+#[allow(dead_code)]
+pub struct RegionQuery {
+    latitude: f32,
+    longitude: f32,
+    // TODO Filters
+}
+
+impl RegionQuery {
+    /// Creates a region query with required fields
+    /// # Arguments
+    /// lat: Latitude in Float format
+    /// long: Longitude in Float format
+    pub fn new(lat: f32, long: f32) -> Self {
+        RegionQuery {
+            latitude: lat,
+            longitude: long,
+        }
+    }
+}
+
+/// Flight Plan Option
+#[derive(Debug, Serialize, Deserialize, ToSchema, Clone)]
+pub struct FlightOption {
+    fp_id: String,
+    vport_depart: String,
+    vport_arrive: String,
+    timestamp: NaiveDateTime,
+}
+
+/// Vertiport Information
+#[derive(Debug, Serialize, Deserialize, ToSchema, Clone)]
+pub struct Vertiport {
+    id: String,
+    #[schema(example = "Mercy Hospital (Public)")]
+    label: String,
+    latitude: f32,
+    longitude: f32,
+}
```

Comment:
We might want to include altitude for more advanced routing capabilities later down the road. In the router lib, nodes also come with the altitude  field, so it might be better to enforce that.

Reply:
This is strictly for data being sent to the customer - they don't need altitude unless the customer's app map is in 3D. The app only needs to know the labels and where to put it on a 2D map

To clarify, svc-cargo is not contacting the router lib. This is the kind of thing I'll add to the SDD:

```mermaid
sequenceDiagram
    participant web as Client App
    participant cargo as svc-cargo
    participant sched as svc-scheduler
    participant pricing as svc-pricing
    participant stor as svc-storage

    stor->>cargo: Vertiport { lat, lon, label }
    cargo->>web: Vertiport {lat, lon, label }
    note over web: Display map of vertiports
    note over web: Customer selects depart. and dest.
    web->>cargo: Dest. and Depart. Vertiport IDs, DateTime
    cargo->>sched: get_itineraries
    sched->>cargo: Itinerary[]
    cargo->>pricing: "select info from itinerary"[]
    pricing->>cargo: pricing data []
    cargo->>web: Itinerary[] + Pricing Data []
    note over web: Make decision on flight
```